docs/articles/ecomm-recys.md

```diff
@@ -0,0 +1,460 @@
+# Creating personalized, real-time recommendation systems - a [notebook](https://github.com/superlinked/superlinked/blob/main/notebook/recommendations_e_commerce.ipynb) article
+
+Pioneered by the likes of Google and AirBnB, embedding-powered recommendation systems have revolutionized recommendation systems by enabling more accurate and personalized recommendations than traditional methods. By representing users and items as high-dimensional vectors in a latent space, embeddings capture similarities and relationships between users and items, and can therefore be used to provide more relevant recommendations. Their compact and dense nature facilitates efficient computation and scalability, which is vital for real-time and large-scale scenarios.
+
+In this article, we'll walk you through how to use the Superlinked library to create an effective RecSys - specifically an e-commerce site that sells mainly clothing - that can be updated in real-time using feedback loops defined by user interactions.
+
+## Achieving personalization despite RecSys vector embedding challenges
+
+While vector embeddings can vastly improve recommendation systems, effectively implementing them requires addressing several challenges, including:
+- **Quality and relevance**: The embedding generation process, architecture, and data must be carefully considered.
+- **Sparse and noisy data**: Embeddings are less effective when they have incomplete or noisy input. Sparse data is the crux of the cold-start problem.
+- **Scalability**: Efficient methods for large datasets are needed; otherwise latency will be an issue.
+
+With Superlinked, you can address these challenges by combining all available data about users and products into rich multimodal vectors. In our e-commerce RecSys example below, we do this using the following Superlinked library elements:
+- **min_max number spaces**: for understanding customer reviews and pricing information
+- **text-similarity space**: for semantic understanding of product information
+- **Events schema** and **Effects** to modify vectors
+- **query time weights** - to define how you want the data to be treated when you run the query, letting you optimize and scale without re-embedding the whole dataset (latency).
+
+By embedding the sparse user-specific data we have - the user's initial product preference - we can handle the cold start problem. As user behavior accrues, we can *hyper*-personalize recommendations by embedding this event data, creating a feedback loop that lets you update vectors with user preferences in real time.
+
+Let's get started.
+
+## Building an e-commerce recommendation engine with Superlinked
+
+At the start of development, we have the following product data:
+- number of reviewers
+- product ratings
+- textual description
+- name of product (usually containing brand name)
+- category
+  
+We have two users (user_1 and user_2) whom we can differentiate by either:
```

Comment:
@morkapronczay
differentiate them by 1 specific on-registration product pref
2 seems like there are two points here: A. in general, users prefer cheaper, more reviewed, higher rated products (so this isn't a way of differentiating them), B. users prefer certain kinds of products (this isn't a way of differentiating user_1 and user_2).

Reply:
yes agreed, better formulation!